sdk/storage/Azure.Storage.Common.DataMovement/src/DirectoryScanner.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Azure.Storage
+{
+    internal class DirectoryScanner
+    {
+        private string _path;
+
+        public DirectoryScanner(string path)
+        {
+            _path = path;
```

Comment:
what if this isn't directory?

Reply:
I think I assumed everything would go through FileSystemScanner in expected manner, but this is a possible case if mocking or just using DirectoryScanner elsewhere without FileSystemScanner. I'll throw an exception if a path isn't a directory.